internal/serviceapi/projectsettingsapi/resource_test.go

```diff
@@ -0,0 +1,108 @@
+package projectsettingsapi_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+)
+
+const resourceName = "mongodbatlas_project_settings_api.test"
+
+func TestAccProjectSettingsAPI_basic(t *testing.T) {
+	var (
+		projectID = acc.ProjectIDExecution(t)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		Steps: []resource.TestStep{
+			{
+				Config: config(projectID, false),
+				Check:  check(projectID, false),
+			},
+			{
+				Config: config(projectID, true),
+				Check:  check(projectID, true),
+			},
+			{
+				Config: config(projectID, false),
+				Check:  check(projectID, false),
+			},
+			{
+				ResourceName:                         resourceName,
+				ImportStateIdFunc:                    importStateIDFunc(resourceName),
+				ImportState:                          true,
+				ImportStateVerify:                    true,
+				ImportStateVerifyIdentifierAttribute: "group_id",
+			},
+		},
+	})
+}
+
+func config(projectID string, enabled bool) string {
+	return fmt.Sprintf(`
+		resource "mongodbatlas_project_settings_api" "test" {
+			group_id = %[1]q
+			is_collect_database_specifics_statistics_enabled = %[2]t
+			is_data_explorer_enabled = %[2]t
+			is_data_explorer_gen_ai_features_enabled = %[2]t
```

Comment:
q: Noticed this is the only attribute that is optional-only, does the API behave differently or we simply missed this additional config?